tdp/core/models/deployment_log.py

```diff
@@ -232,6 +233,63 @@ def from_operations(
                 i += 1
         return deployment_log
 
+    @staticmethod
+    def from_operations_hosts_vars(
+        collections: Collections,
+        operation_host_vars_names: list[tuple[str, Optional[str], Optional[list[str]]]],
+    ) -> "DeploymentLog":
+        """Generate a deployment plan from a list of operations, hosts and extra vars.
+
+        Args:
+            collections: Collections to retrieve the operations from.
+            operation_host_vars_names: List of operations, hosts and extra vars to perform.
+
+        Raises:
+            MissingOperationError: If an operation is a noop.
+            MissingHostForOperationError: If an operation is not found in the collections.
+        """
+        operations_list = []
+        hosts_list = []
+        vars_list = []
+        for operation_names, host_names, var_names in operation_host_vars_names:
+            operations_list.append(operation_names)
+            hosts_list.append(host_names)
+            vars_list.append(var_names)
+
+        collections.check_operations_exist(operations_list)
```

Comment:
In my understanding, this check is already performed within the `collections.check_operations_hosts_exist`.

Reply:
I don't know if I have to remove this test : the first is to create the DeploymentLog object, the second is to create the OperationLog objects.

Reply:
True, here are my arguments:

- If the `collections.check_operations_hosts_exist` raises an error, the `DeploymentLog`` object won't be returned.
- PR [#432](https://github.com/TOSIT-IO/tdp-lib/issues/432) will remove this method 😅 (you can still use the new `collection.check_operation_exist` though).

Reply:
Done, I removed this test